src/main/java/org/apache/pirk/encryption/Paillier.java

```diff
@@ -95,140 +95,185 @@
+   * @throws IllegalArgumentException
+   *           If the {@code certainty} is less than the system allowed lower bound.
    */
-  public Paillier(int bitLengthInput, int certainty) throws PIRException
+  public Paillier(int bitLength, int certainty)
   {
-    this(bitLengthInput, certainty, -1);
+    this(bitLength, certainty, -1);
   }
 
   /**
-   * Constructor to generate keys given the desired bitLength and prime certainty value
+   * Constructs a Paillier algorithm with generated keys and optionally ensures a certain bit is set in the modulus.
    * <p>
-   * Can optionally, ensure a certain bit is set in the modulus (if ensureBitSet != 0)
+   * The generated keys {@code p} and {@code q} will have the given modulus bit length and prime certainty.
    * <p>
-   * The probability that the new BigInteger values represents primes will exceed (1 - (1/2)^certainty). The execution time of this constructor is proportional
-   * to the value of this parameter.
-   *
+   * The probability that the generated keys represent primes will exceed (1 - (1/2)<sup>{@code certainty}</sup>). The execution time of this constructor is
+   * proportional to the value of this parameter.
+   * <p>
+   * When ensureBitSet > -1 the value of bit "{@code ensureBitSet}" in modulus {@code N} will be set.
+   * 
+   * @param bitLength
+   *          The bit length of the resulting modulus {@code N}.
+   * @param certainty
+   *          The probability that the new {@code p} and {@code q} represent prime numbers.
+   * @param ensureBitSet
+   *          index of bit in {@code N} to ensure is set.
+   * @throws IllegalArgumentException
+   *           If the {@code certainty} is less than the system allowed lower bound, or the index of {@code ensureBitSet} is greater than the {@code bitLength}.
    */
-  public Paillier(int bitLengthInput, int certainty, int ensureBitSet) throws PIRException
+  public Paillier(int bitLength, int certainty, int ensureBitSet)
   {
-    bitLength = bitLengthInput;
-
     int systemPrimeCertainty = SystemConfiguration.getIntProperty("pir.primeCertainty", 128);
     if (certainty < systemPrimeCertainty)
     {
-      throw new PIRException("Input certainty = " + certainty + " is less than allowed system lower bound = " + systemPrimeCertainty);
+      throw new IllegalArgumentException("Input certainty = " + certainty + " is less than allowed system lower bound = " + systemPrimeCertainty);
     }
-    if (ensureBitSet >= bitLengthInput)
+    if (ensureBitSet >= bitLength)
     {
-      throw new PIRException("ensureBitSet = " + ensureBitSet + " must be less than bitLengthInput = " + bitLengthInput);
+      throw new IllegalArgumentException("ensureBitSet = " + ensureBitSet + " must be less than bitLengthInput = " + bitLength);
     }
-    generateKeys(certainty, ensureBitSet);
+    this.bitLength = bitLength;
+    generateKeys(bitLength, certainty, ensureBitSet);
```

Comment:
I changed ```generateKeys``` to take ```bitLength``` as a parameter rather than the code relying on the ordering of setting the ```bitLength``` instance variable from the constructor's parameter and calling the method - it just makes ```generateKeys``` self-sufficient should it ever be called from a new constructor/method.  A minor style point I agree.